lib/Backends/OpenCL/kernels.cl

```diff
@@ -1141,6 +1141,56 @@ __kernel void oclavgpoolW(__global void *mem, cl_uint32_t dest, cl_uint32_t src,
   oclavgpoolK(&mem[dest], &mem[src], filterSize, stride, pads, odim, idim);
 }
 
+__kernel void oclavgpool_i8K(__global cl_int8_t *dest, __global cl_int8_t *src,
```

Comment:
why there is this "ocl" prefix?

Reply:
it's the int8 version of kernel oclavgpoolK, and the instruction being executed here is also OCLAvgPoolInst. Not sure why we have both OCLAvgPool and AvgPool? @opti-mix

Reply:
Just talked to @opti-mix about this. Seems like it has to do with the tensor shape (NHWC vs NCHW). Would be good to rename these methods accordingly, if you prefer on a separate cleanup review.

Reply:
sounds cool. But as it would touch more than these files we have in this pr, maybe we'll do in another pr?

Reply:
sounds good.